sdk/src/types/block/semantic/mod.rs

```diff
@@ -145,36 +144,41 @@ impl<'a> SemanticValidationContext<'a> {
                         let account_id = output.account_id_non_null(output_id);
 
                         if self.commitment_context_input.is_none() {
-                            return Ok(Some(TransactionFailureReason::BlockIssuerCommitmentInputMissing));
+                            return Err(TransactionFailureReason::BlockIssuerCommitmentInputMissing);
                         }
                         if !bic_context_inputs.contains(&account_id) {
-                            return Ok(Some(TransactionFailureReason::BlockIssuanceCreditInputMissing));
+                            return Err(TransactionFailureReason::BlockIssuanceCreditInputMissing);
                         }
                         let entry = self.block_issuer_mana.entry(account_id).or_default();
                         entry.0 = entry
                             .0
-                            .checked_add(consumed_output.available_mana(
-                                &self.protocol_parameters,
-                                output_id.transaction_id().slot_index(),
-                                self.transaction.creation_slot(),
-                            )?)
-                            .ok_or(Error::ConsumedManaOverflow)?;
+                            .checked_add(
+                                consumed_output
+                                    .available_mana(
+                                        &self.protocol_parameters,
+                                        output_id.transaction_id().slot_index(),
+                                        self.transaction.creation_slot(),
+                                    )
+                                    // Unwrap is fine as we already checked both slot indices against each others.
+                                    .unwrap(),
```

Comment:
```suggestion
                                    .unwrap(), // Unwrap is fine as we already checked both slot indices against each others.
```

Reply:
I'll keep it as is if you don't mind, we usually put them on the line before